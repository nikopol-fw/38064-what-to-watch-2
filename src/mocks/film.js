export const film = {
  title: `The Grand Budapest Hotel`,
  genre: `Drama`,
  releaseYear: `2014`,
  posterImg: `bg-the-grand-budapest-hotel.jpg`,
  coverImg: `the-grand-budapest-hotel-poster.jpg`,
};
