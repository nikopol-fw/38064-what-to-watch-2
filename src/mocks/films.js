export const films = [{
  title: `Fantastic Beasts: The Crimes of Grindelwald`,
  img: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
}, {
  title: `Bohemian Rhapsody`,
  img: `bohemian-rhapsody.jpg`,
}, {
  title: `Macbeth`,
  img: `macbeth.jpg`,
}, {
  title: `Aviator`,
  img: `aviator.jpg`,
}, {
  title: `We need to talk about Kevin`,
  img: `we-need-to-talk-about-kevin.jpg`,
}];
