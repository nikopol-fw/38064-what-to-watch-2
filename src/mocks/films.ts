import {Film} from "../models/Film";


export const films: Film[] = [{
  backgroundColor: `#A6B7AC`,
  backgroundImage: `https://htmlacademy-react-2.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg`,
  description: `In 1862, Amsterdam Vallon returns to the Five Points area of New York City seeking revenge against Bill the Butcher, his father's killer.`,
  director: `Martin Scorsese`,
  genre: `Crime`,
  id: 1,
  isFavorite: false,
  name: `Gangs of new york`,
  posterImage: `https://htmlacademy-react-2.appspot.com/wtw/static/film/poster/Gangs_of_New_York_Poster.jpg`,
  previewImage: `https://htmlacademy-react-2.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg`,
  previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  rating: 8.8,
  released: 2002,
  runTime: 167,
  scoresCount: 370881,
  starring: [`Leonardo DiCaprio`, `Cameron Diaz`, `Daniel Day-Lewis`],
  videoLink: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
}, {
  backgroundColor: `#9B7E61`,
  backgroundImage: `https://htmlacademy-react-2.appspot.com/wtw/static/film/background/War_of_the_Worlds.jpg`,
  description: `As Earth is invaded by alien tripod fighting machines, one family fights for survival.`,
  director: `Steven Spielberg`,
  genre: `Adventure`,
  id: 2,
  isFavorite: false,
  name: `War of the Worlds`,
  posterImage: `https://htmlacademy-react-2.appspot.com/wtw/static/film/poster/War_of_the_Worlds.jpg`,
  previewImage: `https://htmlacademy-react-2.appspot.com/wtw/static/film/preview/war-of-the-worlds.jpg`,
  previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  rating: 9.3,
  released: 2005,
  runTime: 116,
  scoresCount: 386834,
  starring: [`Tom Cruise`, `Dakota Fanning`, `Tim Robbins`],
  videoLink: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
}, {
  backgroundColor: `#F1E9CE`,
  backgroundImage: `https://htmlacademy-react-2.appspot.com/wtw/static/film/background/Macbeth.jpg`,
  description: `Macbeth, the Thane of Glamis, receives a prophecy from a trio of witches that one day he will become King of Scotland. Consumed by ambition and spurred to action by his wife, Macbeth murders his king and takes the throne for himself.`,
  director: `Justin Kurzel`,
  genre: `Drama`,
  id: 3,
  isFavorite: false,
  name: `Macbeth`,
  posterImage: `https://htmlacademy-react-2.appspot.com/wtw/static/film/poster/Macbeth.jpg`,
  previewImage: `https://htmlacademy-react-2.appspot.com/wtw/static/film/preview/macbeth.jpg`,
  previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  rating: 3.3,
  released: 2015,
  runTime: 113,
  scoresCount: 48798,
  starring: [`Michael Fassbender`, `Marion Cotillard`, `Jack Madigan`],
  videoLink: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
}, {
  backgroundColor: `#92918B`,
  backgroundImage: `https://htmlacademy-react-2.appspot.com/wtw/static/film/background/Revenant.jpg`,
  description: `A frontiersman on a fur trading expedition in the 1820s fights for survival after being mauled by a bear and left for dead by members of his own hunting team.`,
  director: `Alejandro G. Iñárritu`,
  genre: `Action`,
  id: 4,
  isFavorite: false,
  name: `The Revenant`,
  posterImage: `https://htmlacademy-react-2.appspot.com/wtw/static/film/poster/Revenant.jpg`,
  previewImage: `https://htmlacademy-react-2.appspot.com/wtw/static/film/preview/revenant.jpg`,
  previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  rating: 4,
  released: 2015,
  runTime: 156,
  scoresCount: 618498,
  starring: [`Leonardo DiCaprio`, `Tom Hardy`, `Will Poulter`],
  videoLink: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
}, {
  backgroundColor: `#F0DBA2`,
  backgroundImage: `https://htmlacademy-react-2.appspot.com/wtw/static/film/background/Johnny_English.jpg`,
  description: `After a sudden attack on the MI5, Johnny English, Britain's most confident yet unintelligent spy, becomes Britain's only spy.`,
  director: `Peter Howitt`,
  genre: `Comedy`,
  id: 5,
  isFavorite: false,
  name: `Johnny English`,
  posterImage: `https://htmlacademy-react-2.appspot.com/wtw/static/film/poster/Johnny_English.jpg`,
  previewImage: `https://htmlacademy-react-2.appspot.com/wtw/static/film/preview/johnny-english.jpg`,
  previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  rating: 10,
  released: 2003,
  runTime: 88,
  scoresCount: 136843,
  starring: [`Rowan Atkinson`, `John Malkovich`, `Natalie Imbruglia`],
  videoLink: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
}];
