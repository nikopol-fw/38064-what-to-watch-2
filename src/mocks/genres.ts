export const genres = [`All genres`, `Action`, `Adventure`, `Comedy`, `Crime`, `Drama`, `Fantasy`, `Thriller`];
