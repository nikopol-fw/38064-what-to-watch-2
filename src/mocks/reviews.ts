import {Review} from "../models/review";


export const reviews: Review[] = [{
  id: 1,
  rating: 1.3,
  comment: `Poised to be an instant classic, almost everything about this film is phenomenal - the acting, the cinematography, the discography, etc.`,
  date: new Date(`2019-12-05T19:57:19.384Z`),
  user: {
    id: 16,
    name: `Mollie`,
  }
}, {
  id: 2,
  rating: 2.7,
  comment: `This movie really touched my heart, it really is the best movie of the year and everyone should see this masterpiece.`,
  date: new Date(`2019-11-30T19:57:19.384Z`),
  user: {
    id: 12,
    name: `Isaac`,
  }
}];
