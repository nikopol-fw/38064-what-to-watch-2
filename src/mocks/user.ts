import {User} from "../models/user";


export const user: User = {
  avatar: `/wtw/static/avatar/3.jpg`,
  email: `alter.ego@gmail.com`,
  id: 1,
  name: `alter.ego`,
};
