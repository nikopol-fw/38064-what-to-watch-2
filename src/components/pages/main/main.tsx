import * as React from 'react';
import {connect} from 'react-redux';

import {Film} from '../../../models/film';
import {getFilms, getGenres, getPromo} from '../../../reducer/data/selectors';
import Catalog from '../../shared/catalog/catalog';
import {withFilters} from '../../../hocs/with-filters/with-filters';
import {Operation} from '../../../reducer/data/data';
import {User} from '../../../models/user';
import {Header} from '../../shared/header/header';
import {Footer} from '../../shared/footer/footer';
import history from '../../../history';
import {getUserInfo} from '../../../reducer/user/selectors';


const CatalogWrapped = withFilters(Catalog);

interface Props {
  films: Film[];
  genres: string[];

  loadPromo: () => Promise<any>;
  promo: Film;
  setFavorite: (filmId: number, status: (0 | 1)) => Promise<any>;
  user: User;
}

export class MainPage extends React.PureComponent<Props> {

  static readonly defaultProps = {
    films: [],
    genres: [],
  };

  constructor(props: Props) {
    super(props);

    this.clickFavoriteHandler = this.clickFavoriteHandler.bind(this);
    this.clickPlayHandler = this.clickPlayHandler.bind(this);

    props.loadPromo();
  }

  clickFavoriteHandler() {
    this.props.setFavorite(
        this.props.promo.id,
        this.props.promo.isFavorite ? 0 : 1,
    );
  }

  clickPlayHandler() {
    const {promo} = this.props;
    history.push(`/films/${promo.id}/player`);
  }

  render() {
    const {films, genres, promo, user} = this.props;

    return (
      <>
        <section className="movie-card">
          <div className="movie-card__bg">
            <img src={promo && promo.backgroundImage} alt={promo && promo.name}/>
          </div>

          <h1 className="visually-hidden">WTW</h1>

          <Header avatar={user.avatar} isAuth={!!(user && user.id)} name={user.name}/>

          <div className="movie-card__wrap">
            <div className="movie-card__info">
              <div className="movie-card__poster">
                <img src={promo && promo.posterImage} alt={promo && promo.name} width="218" height="327"/>
              </div>

              <div className="movie-card__desc">
                <h2 className="movie-card__title">{promo && promo.name}</h2>
                <p className="movie-card__meta">
                  <span className="movie-card__genre">{promo && promo.genre}</span>
                  <span className="movie-card__year">{promo && promo.released}</span>
                </p>

                <div className="movie-card__buttons">
                  {promo && (
                    <>
                      <button className="btn btn--play movie-card__button" type="button"
                        onClick={this.clickPlayHandler}>
                        <svg viewBox="0 0 19 19" width="19" height="19">
                          <use xlinkHref="#play-s"/>
                        </svg>
                        <span>Play</span>
                      </button>

                      <button className="btn btn--list movie-card__button" type="button"
                        onClick={this.clickFavoriteHandler}>
                        <svg viewBox="0 0 19 20" width="19" height="20">
                          <use xlinkHref={promo.isFavorite ? `#in-list` : `#add`}/>
                        </svg>
                        <span>My list</span>
                      </button>
                    </>
                  )}
                </div>
              </div>

            </div>
          </div>
        </section>


        <div className="page-content">
          <section className="catalog">
            <h2 className="catalog__title visually-hidden">Catalog</h2>

            <CatalogWrapped
              films={films}
              genres={genres}
            />
          </section>
          <Footer/>
        </div>
      </>
    );
  }
}


const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  films: getFilms(state),
  genres: getGenres(state),
  promo: getPromo(state),
  user: getUserInfo(state),
});

const mapDispatchToProps = (dispatch) => ({
  loadPromo: () => dispatch(Operation.loadPromo()),
  setFavorite: (filmId: number, status: (0 | 1)) => dispatch(Operation.setFavoriteStatus(filmId, status)),
});


export default connect(mapStateToProps, mapDispatchToProps)(MainPage);
