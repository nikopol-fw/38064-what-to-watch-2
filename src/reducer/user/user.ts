import {User} from "../../models/user";
import {AxiosInstance, AxiosResponse} from "axios";
import {FormLogin} from "../../models/form-login";
import {LoginApiData} from "../../models/api-login-data";


const initialState = {
  info: {},
};


const ActionType = {
  RESET_USER: `RESET_USER`,
  UPDATE_USER_INFO: `UPDATE_USER_INFO`,
};


const ActionCreator = {
  resetUserInfo: () => ({
    type: ActionType.RESET_USER,
  }),
  updateUserInfo: (userData: User) => ({
    type: ActionType.UPDATE_USER_INFO,
    payload: userData,
  }),
};


const Operation = {
  authorize: (formData: FormLogin) => (dispatch, _getState, api: AxiosInstance) => {
    return api.post(`/login`, formData)
      .then((response: AxiosResponse<LoginApiData>) => {
        if (response && response.status === 200) {
          const data = response.data;
          const UserInfo: User = {
            avatar: data[`avatar_url`],
            email: data[`email`],
            id: data[`id`],
            name: data[`name`],
          };
          dispatch(ActionCreator.updateUserInfo(UserInfo));
        }
      });
  },
};


const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.RESET_USER:
      return Object.assign({}, state, {
        info: initialState.info,
      });

    case ActionType.UPDATE_USER_INFO:
      return Object.assign({}, state, {
        info: action.payload,
      });
  }

  return state;
};


export {
  ActionCreator,
  ActionType,
  Operation,
  reducer,
};
