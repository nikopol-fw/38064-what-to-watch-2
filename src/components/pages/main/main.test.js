import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json';

import {films} from '../../../mocks/films';
import {user} from '../../../mocks/user';
import {MainPage} from './main';


Enzyme.configure({
  adapter: new Adapter(),
});

const mock = {
  films,
  genres: [`All genres`, `Crime`, `Adventure`, `Comedy`],
  loadPromo: () => void (0),
  promo: films[0],
  setFavorite: () => void (0),
  user,
};

it(`MainPage correctly renders after relaunch`, () => {
  const tree = shallow(
      <MainPage
        films={mock.films}
        genres={mock.genres}
        loadPromo={mock.loadPromo}
        promo={mock.promo}
        setFavorite={mock.setFavorite}
        user={mock.user}
      />
  );

  expect(toJSON(tree)).toMatchSnapshot();
});
