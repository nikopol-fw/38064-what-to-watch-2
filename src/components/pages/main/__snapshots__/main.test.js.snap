// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage correctly renders after relaunch 1`] = `
<Fragment>
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="Gangs of new york"
        src="https://htmlacademy-react-2.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <Header
      avatar="/wtw/static/avatar/3.jpg"
      isAuth={true}
      name="alter.ego"
    />
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        >
          <img
            alt="Gangs of new york"
            height="327"
            src="https://htmlacademy-react-2.appspot.com/wtw/static/film/poster/Gangs_of_New_York_Poster.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Gangs of new york
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Crime
            </span>
            <span
              className="movie-card__year"
            >
              2002
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <WithFilters
        films={
          Array [
            Object {
              "backgroundColor": "#A6B7AC",
              "backgroundImage": "https://htmlacademy-react-2.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg",
              "description": "In 1862, Amsterdam Vallon returns to the Five Points area of New York City seeking revenge against Bill the Butcher, his father's killer.",
              "director": "Martin Scorsese",
              "genre": "Crime",
              "id": 1,
              "isFavorite": false,
              "name": "Gangs of new york",
              "posterImage": "https://htmlacademy-react-2.appspot.com/wtw/static/film/poster/Gangs_of_New_York_Poster.jpg",
              "previewImage": "https://htmlacademy-react-2.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg",
              "previewVideoLink": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
              "rating": 8.8,
              "released": 2002,
              "runTime": 167,
              "scoresCount": 370881,
              "starring": Array [
                "Leonardo DiCaprio",
                "Cameron Diaz",
                "Daniel Day-Lewis",
              ],
              "videoLink": "http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4",
            },
            Object {
              "backgroundColor": "#9B7E61",
              "backgroundImage": "https://htmlacademy-react-2.appspot.com/wtw/static/film/background/War_of_the_Worlds.jpg",
              "description": "As Earth is invaded by alien tripod fighting machines, one family fights for survival.",
              "director": "Steven Spielberg",
              "genre": "Adventure",
              "id": 2,
              "isFavorite": false,
              "name": "War of the Worlds",
              "posterImage": "https://htmlacademy-react-2.appspot.com/wtw/static/film/poster/War_of_the_Worlds.jpg",
              "previewImage": "https://htmlacademy-react-2.appspot.com/wtw/static/film/preview/war-of-the-worlds.jpg",
              "previewVideoLink": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
              "rating": 9.3,
              "released": 2005,
              "runTime": 116,
              "scoresCount": 386834,
              "starring": Array [
                "Tom Cruise",
                "Dakota Fanning",
                "Tim Robbins",
              ],
              "videoLink": "http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4",
            },
            Object {
              "backgroundColor": "#F1E9CE",
              "backgroundImage": "https://htmlacademy-react-2.appspot.com/wtw/static/film/background/Macbeth.jpg",
              "description": "Macbeth, the Thane of Glamis, receives a prophecy from a trio of witches that one day he will become King of Scotland. Consumed by ambition and spurred to action by his wife, Macbeth murders his king and takes the throne for himself.",
              "director": "Justin Kurzel",
              "genre": "Drama",
              "id": 3,
              "isFavorite": false,
              "name": "Macbeth",
              "posterImage": "https://htmlacademy-react-2.appspot.com/wtw/static/film/poster/Macbeth.jpg",
              "previewImage": "https://htmlacademy-react-2.appspot.com/wtw/static/film/preview/macbeth.jpg",
              "previewVideoLink": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
              "rating": 3.3,
              "released": 2015,
              "runTime": 113,
              "scoresCount": 48798,
              "starring": Array [
                "Michael Fassbender",
                "Marion Cotillard",
                "Jack Madigan",
              ],
              "videoLink": "http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4",
            },
            Object {
              "backgroundColor": "#92918B",
              "backgroundImage": "https://htmlacademy-react-2.appspot.com/wtw/static/film/background/Revenant.jpg",
              "description": "A frontiersman on a fur trading expedition in the 1820s fights for survival after being mauled by a bear and left for dead by members of his own hunting team.",
              "director": "Alejandro G. Iñárritu",
              "genre": "Action",
              "id": 4,
              "isFavorite": false,
              "name": "The Revenant",
              "posterImage": "https://htmlacademy-react-2.appspot.com/wtw/static/film/poster/Revenant.jpg",
              "previewImage": "https://htmlacademy-react-2.appspot.com/wtw/static/film/preview/revenant.jpg",
              "previewVideoLink": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
              "rating": 4,
              "released": 2015,
              "runTime": 156,
              "scoresCount": 618498,
              "starring": Array [
                "Leonardo DiCaprio",
                "Tom Hardy",
                "Will Poulter",
              ],
              "videoLink": "http://media.xiph.org/mango/tears_of_steel_1080p.webm",
            },
            Object {
              "backgroundColor": "#F0DBA2",
              "backgroundImage": "https://htmlacademy-react-2.appspot.com/wtw/static/film/background/Johnny_English.jpg",
              "description": "After a sudden attack on the MI5, Johnny English, Britain's most confident yet unintelligent spy, becomes Britain's only spy.",
              "director": "Peter Howitt",
              "genre": "Comedy",
              "id": 5,
              "isFavorite": false,
              "name": "Johnny English",
              "posterImage": "https://htmlacademy-react-2.appspot.com/wtw/static/film/poster/Johnny_English.jpg",
              "previewImage": "https://htmlacademy-react-2.appspot.com/wtw/static/film/preview/johnny-english.jpg",
              "previewVideoLink": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
              "rating": 10,
              "released": 2003,
              "runTime": 88,
              "scoresCount": 136843,
              "starring": Array [
                "Rowan Atkinson",
                "John Malkovich",
                "Natalie Imbruglia",
              ],
              "videoLink": "http://media.xiph.org/mango/tears_of_steel_1080p.webm",
            },
          ]
        }
        genres={
          Array [
            "All genres",
            "Crime",
            "Adventure",
            "Comedy",
          ]
        }
      />
    </section>
    <Footer />
  </div>
</Fragment>
`;
